probe-rs/src/architecture/arm/traits.rs

```diff
@@ -91,19 +93,29 @@ pub trait DapAccess {
     ///
     /// Highest 4 bits of `addr` are interpreted as the bank number, implementations
     /// will do bank switching if necessary.
-    fn read_raw_dp_register(&mut self, addr: u8) -> Result<u32, DebugProbeError>;
+    fn read_raw_dp_register(&mut self, dp: Option<u32>, addr: u8) -> Result<u32, DebugProbeError>;
```

Comment:
I'm not a big fan of `Option` in function arguments, I think it's not easy to understand what happens when you look at function calls:
```rust
dp.read_raw_dp_register(None, 4);
dp.read_raw_dp_register(Some(1), 4);
```
I would prefer to solve this with an enum, e.g.:
```rust
dp.read_raw_dp_register(Dp::Default, 4);
dp.read_raw_dp_register(Dp:Dp(1), 4);
```

Reply:
Makes sense! Then the enum variants can get in-depth docs. Will do

Reply:
Done!